.github/workflows/build_servers.yml

```diff
@@ -0,0 +1,90 @@
+name: Build Servers from a Pull Request
+# build_servers.yml
+
+on:
+  pull_request:
+    paths:
+      # This workflow should run when a file in the src/ directory has been modified.
+      - 'src/**'
+      - '3rdparty/**'
+
+jobs:
+  build:
+    # Github Actions checks for '[ci skip]', '[skip ci]', '[no ci]', '[skip actions]', or '[actions skip]' but not a hyphenated version.
+    # It's a catch-all incase a Pull Request has been opened and someone is on auto-pilot.
+    if: "!contains(github.event.head_commit.message, 'ci-skip')"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+          # The ubuntu-latest label currently points to ubuntu-18.04.
+          # Available: ubuntu-20.04, ubuntu-18.04	
+          os: [ubuntu-latest]
+          # There is no reason to check build success for packet-versions older than four years.
+          packetver: ['20211103', '20200902', '20200401', '20180620']
```

Comment:
As discussed please put this in a separate action.
No need to compile all these on all compilers.